lib/service_manager.dart

```diff
@@ -139,114 +129,75 @@ class IsolateManager {
-      if (_flutterIsolate == null) {
-        if (_isFlutterExtension(event.extensionRPC)) {
-          _setFlutterIsolate(event.isolate);
-        }
+      // Check to see if there is a new isolate.
+      if (_selectedIsolate == null) {
+        _setSelectedIsolate(event.isolate);
       }
     } else if (event.kind == 'IsolateExit') {
       _isolates.remove(event.isolate);
-      _flutterIsolates.remove(event.isolate);
       _isolateExitedController.add(event.isolate);
       if (_selectedIsolate == event.isolate) {
         _selectedIsolate = _isolates.isEmpty ? null : _isolates.first;
         _selectedIsolateController.add(_selectedIsolate);
-      }
-      if (_flutterIsolate == event.isolate) {
-        _setFlutterIsolate(
-            _flutterIsolates.isEmpty ? null : _flutterIsolates.first);
         _serviceExtensionManager.resetAvailableExtensions();
       }
     }
   }
 
-  IsolateRef _selectBestFirstIsolate(List<IsolateRef> isolates) {
-    final IsolateRef ref = isolates.firstWhere((IsolateRef ref) {
-      // 'foo.dart:main()'
-      return ref.name.contains(':main(');
-    }, orElse: () => null);
-
-    if (ref != null) {
-      return ref;
-    }
-
-    return isolates.isEmpty ? null : isolates.first;
-  }
-
-  bool _isFlutterExtension(String extensionName) {
-    return extensionName.startsWith('ext.flutter.');
-  }
-
-  Future<void> _initFlutterIsolate(List<IsolateRef> isolates) async {
+  Future<void> _selectBestFirstIsolate(List<IsolateRef> isolates) async {
     for (IsolateRef ref in isolates) {
-      // Populate flutter isolate info.
-      if (_flutterIsolate == null) {
-        final Isolate isolate = await _service.getIsolate(ref.id);
-        if (isolate.extensionRPCs != null) {
-          for (String extensionName in isolate.extensionRPCs) {
-            if (_isFlutterExtension(extensionName)) {
-              _setFlutterIsolate(ref);
-              break;
-            }
-          }
+      if (_selectedIsolate == null) {
+        if (ref.name.contains(':main(')) {
+          _selectedIsolate = ref;
+          break;
         }
       }
-      // On initial connection to running app, service extensions are added from
-      // here.
-      await _serviceExtensionManager._addRegisteredExtensionRPCs(ref);
     }
+    _selectedIsolate ??= isolates.isNotEmpty ? isolates.first : null;
   }
 
-  void _setFlutterIsolate(IsolateRef ref) {
-    if (_flutterIsolate == ref) {
-      // Isolate didn't change. Do nothing.
+  void _setSelectedIsolate(IsolateRef ref) {
+    if (_selectedIsolate == ref) {
       return;
     }
-    _flutterIsolate = ref;
-    _flutterIsolateController.add(ref);
+    _selectedIsolate = ref;
+    _selectedIsolateController.add(ref);
   }
 
-  StreamSubscription<IsolateRef> getCurrentFlutterIsolate(
+  StreamSubscription<IsolateRef> getSelectedIsolate(
       void onData(IsolateRef ref)) {
-    if (_flutterIsolate != null) {
-      onData(_flutterIsolate);
+    if (_selectedIsolate != null) {
```

Comment:
I think the _flutterIsolate code is more robust than the _selectedIsolate code. I would keep that version instead.
ref.name.contains(':main(' . seems like the worse hack.

Reply:
Done. Using the flutterIsolate code. Left the naming as selectedIsolate because I think that makes sense with the other isolate manager code.